src/impact-dashboard/EpidemicModelContext.tsx

```diff
@@ -148,93 +186,166 @@ function epidemicModelReducer(
       // then submit it all at once; therefore we can just expose a single
       // action and merge the whole object into previous state.
       // it's not very granular but it doesn't need to be at the moment
-      return Object.assign({}, state, action.payload);
-    case "reset":
-      console.log(action.payload);
-      return getResetState(action.payload);
+
+      // change in state or county triggers a bigger reset
+      let updates = { ...action.payload };
+      let { stateCode, countyName, countyLevelData } = updates;
+
+      countyLevelData = countyLevelData || state.countyLevelData;
+      if (countyLevelData) {
+        if (stateCode) {
+          countyName = countyName || "Total";
+        } else if (countyName) {
+          stateCode = state.stateCode;
+        }
+        if (stateCode && countyName) {
+          return Object.assign(
+            getResetBase(stateCode, countyName),
+            getLocaleData(countyLevelData, stateCode, countyName),
+          );
+        }
+      }
+      return Object.assign({}, state, updates);
   }
 }
 
+function sanitizeQueryParams(rawQueryParams: QueryParams) {
+  return mapValues(pick(rawQueryParams, urlParamKeys), (value) => {
+    // most of these are numbers but some are strings
+    const n = numeral(value).value();
+    return n != null ? n : "" + value;
```

Comment:
Plus sign concatenation in JS always freaks me out because I'm never sure what I'm gonna get. Is this canonical, or should we do `str(value)`?